mod_ci/controllers.py

```diff
@@ -905,6 +907,29 @@ def blocked_users():
             g.db.add(blocked_user)
             g.db.commit()
             flash('User blocked successfully.')
+            # Remove any queued pull request from blocked user
+            gh = GitHub(access_token=g.github['bot_token'])
+            repository = gh.repos(g.github['repository_owner'])(g.github['repository'])
+            try:
+                # Getting all pull requests by blocked user on the repo
+                pulls = requests.get('https://api.github.com/repos/ccextractor/sample-platform/pulls/head={username}'
+                                     ).format(username=usernames[addUserForm.userID.data])
+                for pull in pulls:
+                    test = Test.query.filter(Test.pr_nr == pull['number']).all()
+                    # Add canceled status only if the test hasn't started yet
+                    if len(test.progress) > 0:
+                        continue
+                    progress = TestProgress(test.id, TestStatus.canceled, "PR closed", datetime.datetime.now())
+                    g.db.add(progress)
+                    g.db.commit()
+                    repository.statuses(test.commit).post(
+                        state=Status.FAILURE,
+                        description="Tests canceled since user blacklisted",
+                        context="CI - {name}".format(name=test.platform.value),
+                        target_url=url_for('test.by_id', test_id=test.id, _external=True)
+                    )
+            except requests.exceptions.RequestException:
+                log.error('Pull Requests of Blocked User could not be fetched.')
```

Comment:
You have the context available, so you can use g.log instead. That also makes you don't need to import it at the top.